lib/integrations/awsoidc/deployservice_iam_config.go

```diff
@@ -215,44 +197,7 @@ func ConfigureDeployServiceIAM(ctx context.Context, clt DeployServiceIAMConfigur
 	return nil
 }
 
-// createBoundaryPolicyForTaskRole creates a Policy to be used as TaskRole's Role Boundary.
-// It allows for the TaskRole to:
-// - connect to any RDS DB
-// - Get, Put and Delete Role Policies to manage the Policy Statements when adding other rds-db:connect entries
-// - write application logs to CloudWatch
-func createBoundaryPolicyForTaskRole(ctx context.Context, clt DeployServiceIAMConfigureClient, req DeployServiceIAMConfigureRequest) error {
-	taskRoleARN := awslibutils.RoleARN(req.partitionID, req.AccountID, req.TaskRole)
-
-	taskRoleBoundaryPolicyDocument, err := awslib.NewPolicyDocument(
-		awslib.StatementForRDSDBConnect(),
-		awslib.StatementForIAMEditRolePolicy(taskRoleARN),
-		awslib.StatementForWritingLogs(),
-	).Marshal()
-	if err != nil {
-		return trace.Wrap(err)
-	}
-
-	_, err = clt.CreatePolicy(ctx, &iam.CreatePolicyInput{
```

Comment:
> // The following actions must be allowed by the IAM Role assigned in the Client.                                
// - iam:CreatePolicy                                
// - iam:CreateRole                                
// - iam:PutRolePolicy                                
// - iam:TagPolicy                                
// - iam:TagRole   

 The CreatePolicy is no longer needed?

Reply:
updated comment to drop iam:CreatePolicy, iam:TagPolicy, and removed the interface CreatePolicy method / test mock impl since those are not needed anymore.

Reply:
diff: https://github.com/gravitational/teleport/compare/f992c79376682f6f5ca272fe204b76642b73a658..1dcc4828b8516e5e311477dc8994c7bd2b872710